### ISSUE TITLE
FOXML fixture loader

### ISSUE THREAD

**Issue by cbeer** [fcrepo4-archive/ff-modeshape-prototype#5]
It'd be so nice to push FOXML into Modeshape and have it deserialized into modeshape objects. Lots of the integration tests I have available to me want to upload a bunch of FOXML as fixtures.

**Comment by ajs6f** [comment]
Sounds like a job for... (ducks into 'phone booth):

MODESHAPE SEQUENCERS!

https://docs.jboss.org/author/display/MODE/Sequencing

I'll get on it. I don't relish the job, and this is one where I may ask to share the pain.

**Comment by cbeer** [comment]
+1; I saw the sequencer docs and thought they looked like just the ticket. who knows how painful it'll be to work with (let alone work with FOXML :/ )

**Comment by ajs6f** [comment]
I wouldn't be surprised if FOXML was more trouble than sequencing. In any event, I'm turning to this one, and I'll report via email as soon as I have anything. This will be the first work I do "en branche".

**Comment by ajs6f** [comment]
On pause, waiting for responses here:

https://community.jboss.org/thread/220261

**Comment by ajs6f** [comment]
We made it. Here:

https://github.com/futures/ff-modeshape-prototype/tree/FOXML

is the first draft. It only loads actually content for X-type datastreams and doesn't yet do an lot of what we will want, because I wanted to get this back to the trunk before wandering too far. Even so, it could be a bear to merge. I had to alter the ModeShape repo config, and we're going to be giving ModeShape two bug reports, I think. Still, we have SEQUENCING! I wrote the parser itself in Scala, because it's so much faster to do stuff like that.
